### PR TITLE
Add try catch to handle local storage errors

### PR DESCRIPTION
**Pull request by dannytranlx**
@lemieux @jugarrit @jpjoyal @spasiu 

Fixes the issue crashing Safari in private mode since the localStorage is disabled. It is showing up in `window`, throws when we try to `setItem`.

### CODE REVIEW

File: src/js/controllers/chatController.js [hunk 1]
```diff
@@ -331,14 +331,21 @@ module.exports = ViewController.extend({
...
+            latestReadTs = 0;
         }
         return latestReadTs;
     },
 
     _setLatestReadTime: function(ts) {
         var key = this._getLatestReadTimeKey();
-        localStorage.setItem(key, ts);
+
+        try {
```
Comment (alavers) [inline src/js/controllers/chatController.js]: Might be worth wrapping localStorage.setItem in a util method so you don't have this duplicated code

**Comment by alavers** [comment]
Rest looks :+1:

File: src/js/main.js [hunk 2]
```diff
@@ -5,13 +5,14 @@ var Backbone = require('backbone');
 var _ = require('underscore');
 var $ = require('jquery');
 var uuid = require('uuid');
-var urljoin = require('url-join');
+var urljoin = require('urljoin');
```
Comment (dannytranlx) [inline src/js/main.js]: @lemieux This is weird, how did your build pass before?

**Comment by jpjoyal** [comment]
:+1:

File: src/js/main.js (hunk 2 above)
Comment (lemieux) [inline src/js/main.js]: very good question

File: src/js/utils/localStorage.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+function setItem(key, value) {
+    try {
+        localStorage.setItem(key, value);
```
Comment (lemieux) [inline src/js/utils/localStorage.js]: can you just fallback to an in memory object?

Comment (dannytranlx) [inline src/js/utils/localStorage.js]: Done!

**Comment by lemieux** [comment]
:+1:

**Pull request merged**